lib/bolt/catalog.rb

```diff
@@ -58,6 +58,8 @@ def compile_catalog(request)
       target = request['target']
       pdb_client = Bolt::PuppetDB::Client.new(Bolt::PuppetDB::Config.new(request['pdb_config']))
       options = request['puppet_config'] || {}
+      project = (request['project'] || {})
+      bolt_project = Struct.new(:name, :path).new(project['name'], project['path']) unless project.empty?
```

Comment:
I think the actual fix here is that `Bolt::Project.new` should just take data and not try to load a file. Loading the file should happen via a factory method which then passes the data to the constructor. Then this could just be `Bolt::Project.new(...)`.

Reply:
I thought about doing that as part of #1842 - it'd also make testing Projects a lot easier.